src/InfectionApplication.php

```diff
@@ -248,6 +259,13 @@ private function getBadMsiErrorMessage(MetricsCalculator $metricsCalculator): st
             );
         }
 
+        throw new \LogicException('Seems like something is wrong with calculations and min-msi options.');
+    }
+
+    private function getBadCoveredMsiErrorMessage(MetricsCalculator $metricsCalculator): string
+    {
+        $baseMessage = 'The minimum required %s percentage should be %s%%, but actual is %s%%. Improve your tests!';
```

Comment:
Could you please reduce code duplication a little bit:

1. This message probably can be moved to const and reused on line 251
2. and there is a duplicated exception message `throw new \LogicException('Seems like something is wrong with calculations and min-msi options.');`. I suggest to create a custom one with incapsulated message or reuse this message somehow in this class

Reply:
I've now extracted the `$baseMessage` into a constant and created a new `MsiCalculationException`.